File: Solutions/3301-3400/3345_Smallest_Divisible_Digit_Product_I/C/Solution01.c
int smallestNumber(int n, int t) {
    int digit1 = 0, digit0 = 0;
    while(1){
        digit1 = n  >= 100 ? n / 100 : n >= 10 ? n / 10 : 1;
        digit0 = n % 10;
        if ((digit1 * digit0) % t == 0) return n;
        n++;
    }
    return 0;
}

File: Solutions/3301-3400/3345_Smallest_Divisible_Digit_Product_I/Python3/Solution01.py
class Solution:
    def smallestNumber(self, n: int, t: int) -> int:
        digit1 = 0; digit0 = 0

        while True:
            if n >= 100 : digit1 = n // 100
            elif n >= 10 : digit1 = n // 10
            else : digit1 = 1
            digit0 = n % 10

            if (digit1 * digit0 % t == 0): return n
            n += 1